ui/window.go

```diff
@@ -1238,9 +989,49 @@ func NewWindow(doRestart chan bool, app *tview.Application, session *discordgo.S
 
 	window.SwitchToGuildsPage()
 
+	app.SetFocusDirectionHandler(tview.Up, shortcuts.FocusUp.Equals)
+	app.SetFocusDirectionHandler(tview.Down, shortcuts.FocusDown.Equals)
+	app.SetFocusDirectionHandler(tview.Left, shortcuts.FocusLeft.Equals)
+	app.SetFocusDirectionHandler(tview.Right, shortcuts.FocusRight.Equals)
+
+	window.messageInput.internalTextView.SetNextFocusableComponents(tview.Up, window.chatView.internalTextView)
```

Comment:
This seems like too tight a coupling to me. The `window` knows

> Component B is focussed. Left is component A and right is component C

Rather than telling each component about the others, could you have the `window` respond to the key events and set focus itself? E.g.

```go
func OnKeyEvent(evt tcell.KeyEvent){
    switch evt.Key {
        case tcell.Left:
            moveFocusLeft()
    }
}

func moveFocusLeft(){
    switch focussedComponent {
        case window.messageInput: window.chatView.SetFocus(window.app)
        case window.chatView: window.guildList.SetFocus(window.app)
    }
}
```

It'll end up being a similarly sized block of code, but the information `window` has will stay with it, instead of spreading out to the internals of the other components

Reply:
I wanted to keep it simple. I am unsure whether there's any benefit in preventing the coupling.
My initial plan was to built automatic focus switching into tview. This is very common in most GUI frameworks, but it doesn't quite suite the needs I have, as most other frameworks only have Forward and Backward focus, but not travel in 4 directions.

So, the way it is done now, the focusing itself happens in tview and the components themselves do not know which components will be focused next. Instead, the window tells them "hey, if anyone asks you, tell them to focus this". So I don't think the components are tightly coupled that way. I didn't have to change any code in the components themselves.

Reply:
Your way would have one benefit though. Right now there's still a decision necessary, which is why I've done parts of the focus decisions in `SwitchToXXXPage`. Maybe I should instead allow to set multiple input handlers. I kinda dislike the fact that there's one big global keyboard handler. You make one mistake in it and everything is f'd up. So i thought the way I did it was a bit safer at least.

Reply:
I'll be merging this for now. While your approach would make things more error prone, I don't think it's too important right now. I'll be making a follow-up issue on this.